dot/node_integration_test.go

```diff
@@ -0,0 +1,329 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+//go:build integration
+// +build integration
+
+package dot
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/core"
+	"github.com/ChainSafe/gossamer/dot/state"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/babe"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/ChainSafe/gossamer/lib/grandpa"
+	"github.com/ChainSafe/gossamer/lib/keystore"
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInitNode(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genFile := NewTestGenesisRawFile(t, cfg)
+
+	cfg.Init.Genesis = genFile
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+}
+
+func TestInitNode_GenesisSpec(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genFile := newTestGenesisFile(t, cfg)
+
+	cfg.Init.Genesis = genFile
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+}
+
+func TestNodeInitialized(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genFile := NewTestGenesisRawFile(t, cfg)
+
+	cfg.Init.Genesis = genFile
+
+	expected := NodeInitialized(cfg.Global.BasePath)
+	require.Equal(t, expected, false)
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	expected = NodeInitialized(cfg.Global.BasePath)
+	require.Equal(t, expected, true)
+}
+
+func TestNewNode(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genFile := NewTestGenesisRawFile(t, cfg)
+
+	cfg.Init.Genesis = genFile
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	ks := keystore.NewGlobalKeystore()
+	err = keystore.LoadKeystore("alice", ks.Gran)
+	require.NoError(t, err)
+	err = keystore.LoadKeystore("alice", ks.Babe)
+	require.NoError(t, err)
+
+	cfg.Core.Roles = types.FullNodeRole
+
+	node, err := NewNode(cfg, ks)
+	require.NoError(t, err)
+
+	bp := node.Services.Get(&babe.Service{})
+	require.NotNil(t, bp)
```

Comment:
Can you assert it matches an expected babe service instead of only not-nil?

Same suggestion for tests below

Reply:
Yes, changed to require.IsType check.

Reply:
Well I meant to have an assertion against an expected `&base.Service{...}`, not just the type check